packages/store/ts/codegen/field.ts

```diff
@@ -82,27 +82,47 @@ export function renderFieldMethods(options: RenderTableOptions) {
       const dynamicSchemaIndex = schemaIndex - options.staticFields.length;
 
       if (options.withGetters) {
-        result += renderWithFieldSuffix(options.withSuffixlessFieldMethods, field.name, (_methodNameSuffix) =>
-          renderWithStore(
-            storeArgument,
-            ({ _typedStore, _store, _commentSuffix, _methodNamePrefix }) => `
+        if (field.typeWrappingData && field.typeWrappingData.kind === "staticArray") {
+          result += renderWithFieldSuffix(options.withSuffixlessFieldMethods, field.name, (_methodNameSuffix) =>
+            renderWithStore(
+              storeArgument,
+              ({ _typedStore, _commentSuffix, _methodNamePrefix }) => `
               /**
                * @notice Get the length of ${field.name}${_commentSuffix}.
                */
               function ${_methodNamePrefix}length${_methodNameSuffix}(${renderArguments([
-              _typedStore,
-              _typedTableId,
-              _typedKeyArgs,
-            ])}) internal view returns (uint256) {
-                ${_keyTupleDefinition}
-                uint256 _byteLength = ${_store}.getDynamicFieldLength(_tableId, _keyTuple, ${dynamicSchemaIndex});
-                unchecked {
-                  return _byteLength / ${portionData.elementLength};
-                }
+                _typedStore,
+                _typedTableId,
+                _typedKeyArgs,
+              ])}) internal pure returns (uint256) {
+                return ${field.typeWrappingData?.staticLength};
               }
           `
-          )
-        );
+            )
+          );
+        } else {
+          result += renderWithFieldSuffix(options.withSuffixlessFieldMethods, field.name, (_methodNameSuffix) =>
+            renderWithStore(
+              storeArgument,
+              ({ _typedStore, _store, _commentSuffix, _methodNamePrefix }) => `
+            /**
+             * @notice Get the length of ${field.name}${_commentSuffix}.
+             */
+            function ${_methodNamePrefix}length${_methodNameSuffix}(${renderArguments([
+                _typedStore,
+                _typedTableId,
+                _typedKeyArgs,
+              ])}) internal view returns (uint256) {
+              ${_keyTupleDefinition}
+              uint256 _byteLength = ${_store}.getDynamicFieldLength(_tableId, _keyTuple, ${dynamicSchemaIndex});
+              unchecked {
+                return _byteLength / ${portionData.elementLength};
+              }
+            }
+        `
+            )
```

Comment:
```suggestion
              ({ _typedStore, _store, _commentSuffix, _methodNamePrefix }) => `
                /**
                 * @notice Get the length of ${field.name}${_commentSuffix}.
                 */
                function ${_methodNamePrefix}length${_methodNameSuffix}(${renderArguments([
                    _typedStore,
                    _typedTableId,
                    _typedKeyArgs,
                  ])}) internal view returns (uint256) {
                  ${_keyTupleDefinition}
                  uint256 _byteLength = ${_store}.getDynamicFieldLength(_tableId, _keyTuple, ${dynamicSchemaIndex});
                  unchecked {
                    return _byteLength / ${portionData.elementLength};
                  }
                }
            `
            )
```
just aligning whitespace